plugin/src/main/java/org/opensearch/ml/sdkclient/LocalClusterIndicesClient.java

```diff
@@ -102,30 +113,55 @@ public CompletionStage<GetDataObjectResponse> getDataObjectAsync(GetDataObjectRe
                     .parser(Optional.of(parser))
                     .source(getResponse.getSource())
                     .build();
-            } catch (OpenSearchStatusException | IndexNotFoundException notFound) {
-                throw notFound;
-            } catch (Exception e) {
-                throw new OpenSearchException(e);
+            } catch (IOException e) {
```

Comment:
so for other unchecked exceptions, it will automatically throw the exception what we have in SdkClient?

Reply:
@dhrubo-os Correct, any other exception will be a non-null `Throwable` supplied to the completable future.  It's tested in `LocalClusterIndicesClientTests.testGetDataObject_Exception()` passing an `UnsupportedOperationException` that would bypass this catch block.